### PR TITLE
[Concept Entry] AI  Diffusion Models #2109

### PR DESCRIPTION
**Pull request by DevGat718**
### Description

 creating a entry on AI diffusion models as a contribution to docs.

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by DevGat718** [comment]
getting the party started aiming to have this done in a week or 2

**Comment by DevGat718** [comment]
@SSwiniarski done with the initial overview.

**Comment by DevGat718** [comment]
yall on vacation? @SSwiniarski @yangc95

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai diffusion model example.png
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai diffusion model example.png:1]: This image needs to be relocated and renamed. The image goes in the media/ folder, and should not have spaces in the filename. The instructions for properly adding an image to an entry are in the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#file-hosting).

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:7]:
```suggestion
```
Large language model is not one of the subjects in the [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) file.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:23]:
```suggestion
## About Diffusion Models

```
Blank line after header

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:36]:
```suggestion
A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.

The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples.

- **DDPMs** : Denoising Diffusion Probabilistic Models are generative models mainly used to remove noise from visual or audio data.
- **SGMs** : Noise-Conditioned Score-Based Generative Models can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution. In English this means building of an estimate based on observed data of which we would like to sample.
- **SDEs** : Stochastic Differential Equations describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes.

```
Reformatted this as a single bulleted list.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:37]:
```suggestion
## Image Generation

```

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:38]: Review the instructions on adding an image in the style guide [here](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#file-hosting).

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:40]:
```suggestion
```
Remove this header.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@DevGat718 I did a first review and left a few comments.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:7]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:23]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:36]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:37]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:38]: copy

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:40]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: forgot to add it, can I add it in? @SSwiniarski

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:34]: From the instructions on file hosting in the style guide:

> Images should be linked to using the following url: `https://raw.githubusercontent.com/Codecademy/docs/main/media/` followed by the image name.

You also need to put something descriptive in the alt text.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:33]:
```suggestion

```
You still need to get rid of this header.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:22]:
```suggestion
### About Diffusion Models

```
You still need a blank line after this header.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [inline content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:32]:
```suggestion
A Diffusion model is a deep learning generative model that works by adding noise to the available training data known as the forward diffusion process, it then reverses the process to recover the data. This is known as denoising or the reverse diffusion process.

The three fundamental mathematical frameworks that support the science behind diffusion models, work on the same principles of adding noise and removing it to generate new samples. 

- **DDPMs** : Denoising Diffusion Probabilistic Models (generative models mainly used to remove noise from visual or audio data).
- **SGMs** : Noise-Conditioned Score-Based Generative Models (can generate new samples from a given distribution. They work by learning an estimation score function that can estimate the log density of the target distribution). In english this means buiding of an estimate based on observed data of which we would like to sample.
- **SDEs** : Stochastic Differential Equations (describe changes in random processes concerning time. Widely used in physics and financial markets involving random factors that significantly impact market outcomes).

```
Please follow all the edits in the suggestion. (You can just click on the commit suggestion button.)

**Review by SSwiniarski — COMMENTED** [review]
@DevGat718 I went through again and saw you missed a few suggestions.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:33]: I had 2 headers with the same name I deleted the other one, there was no edit for this one. Still delete it?

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: this was the change

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:33]: this was the deletion from rd 1

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:34]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:22]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:32]: done

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: I don't see a /main folder to upload too, or maybe im just going about it wrong with the linking

Comment (SSwiniarski) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: @DevGat718 you upload to the /media folder. That URL above is the URL the image will have after it is merged into the main Docs repo. You can't have a relative link because the URL of the entry when it's on the Doc's website is going to be different than the source on GitHub, but the image is still going to live on the GitHub repo.

Comment (DevGat718) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md]: copy I put it in media & put in that link lmk if its good.

File: content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md
Comment (SSwiniarski) [reply content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md:33]: Yes.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]

**Comment by DevGat718** [comment]
done.

**Review by SSwiniarski — APPROVED** [review]
@DevGat718 I did a few minor edits (And I put the header back, I was wrong) and this looks good for a second review. One note is you should change the path of this from `content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md` to `content/ai/concepts/models/models/ai-diffusion-models/ai-diffusion-models.md`

**Comment by DevGat718** [comment]
path changed @SSwiniarski

**Review by KTom101 — CHANGES_REQUESTED** [review]
@DevGat718 I've completed a second review and added some minor changes.

File: content/ai/concepts/foundation-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:20]: done

File: content/ai/concepts/foundation-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:29]: done

File: content/ai/concepts/foundation-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:30]: done

File: content/ai/concepts/foundation-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:28]: done

File: content/ai/concepts/foundation-models/ai-diffusion-models.md
Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: Hey, 

Where are you guys getting these path changes from? the original path was (docs/content/ai/concepts/ai-models/models/ai-diffusion-models/ai-diffusion-models.md) which I got directly from the entry.

Why does it keep changing? 
@KTom101

Comment (KTom101) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: @DevGat718 Thank you for pointing this out and apologies for the confusion! I'm going to verify with the team to make sure the path in the issue was written correctly. 

`ai-models` doesn't currently exist and term entries are usually placed in a `terms` folder, not a `models` folder. Without the `terms` folder, it causes an error in the "test" check which you can see at the bottom of the [first page](https://github.com/Codecademy/docs/pull/2534).

Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: ok copy, 

Thanks for the clarity, so im going with the path (content/ai/concepts/foundation-models/terms/ai-diffusion-models/ai-diffusion-models.md.) now?

Comment (KTom101) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: @DevGat718 Apologies for the delay. I was waiting on our AI expert to get back to me to ensure we weren't trying to make a new concept here. However, I took a closer look at it myself and I feel we should go with the `content/ai/concepts/foundation-models/terms/ai-diffusion-models/ai-diffusion-models.md` path. 

Can you also run `npm run format:all` while making the change to fix the other two checks as well?

Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: ok copy.

Comment (DevGat718) [reply content/ai/concepts/foundation-models/ai-diffusion-models.md:1]: done

**Comment by KTom101** [comment]
@DevGat718 Can we also remove the media file `media/.html-video-tag-example.mp4.icloud`? Was there a reason for it?

**Comment by DevGat718** [comment]
> @DevGat718 Can we also remove the media file `media/.html-video-tag-example.mp4.icloud`? Was there a reason for it?

yeah I can delete it idk where that came from honestly I only added a image @KTom101

**Comment by KTom101** [comment]
@DevGat718 It looks like you were trying to make a change to the media in commit [ae06dc8](https://github.com/Codecademy/docs/pull/2534/commits/ae06dc88cdf3b7bae5bfe4f93380a6d75547e507) . There was a deletion and an insertion there.

**Comment by DevGat718** [comment]
> @DevGat718 It looks like you were trying to make a change to the media in commit [ae06dc8](https://github.com/Codecademy/docs/pull/2534/commits/ae06dc88cdf3b7bae5bfe4f93380a6d75547e507) . There was a deletion and an insertion there.

Yeah I saw it & deleted it. I thought it went through idk where that even came from but up popped up highlighted in vs code.

**Comment by DevGat718** [comment]
@KTom101 should be good now.

**Review by KTom101 — APPROVED** [review]
@DevGat718 I made some minor changes directly and ran `npm run format:all` to fix the [failing workflows](https://www.codecademy.com/resources/docs/contribution-guide/getting-set-up-on-your-computer#:~:text=continuous%20integration%20(CI)%20workflows). Otherwise, LGTM.

**Comment by KTom101** [comment]
@DevGat718 Congrats on the new entry https://www.codecademy.com/resources/docs/ai/foundation-models/diffusion-models

Screenshot 2023-07-25 at 5 00 00 PM

cc: @yangc95

**Comment by DevGat718** [comment]
yay! thanks

**Pull request merged**
